src/status_im/contexts/profile/edit/name/view.cljs

```diff
@@ -0,0 +1,66 @@
+(ns status-im.contexts.profile.edit.name.view
+  (:require [clojure.string :as string]
+            [quo.core :as quo]
+            [react-native.core :as rn]
+            [react-native.safe-area :as safe-area]
+            [reagent.core :as reagent]
+            [status-im.constants :as constants]
+            [status-im.contexts.profile.edit.name.style :as style]
+            [status-im.contexts.profile.edit.name.utils :as utils]
+            [status-im.contexts.profile.utils :as profile.utils]
+            [utils.debounce :as debounce]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn view
+  []
+  (let [insets              (safe-area/get-insets)
+        profile             (rf/sub [:profile/profile-with-image])
+        customization-color (rf/sub [:profile/customization-color])
+        display-name        (profile.utils/displayed-name profile)
+        full-name           (reagent/atom display-name)
+        error-msg           (reagent/atom nil)
+        validate-name       (debounce/debounce #(reset! error-msg
+                                                  (utils/validation-name %))
+                                               500)
+        on-change-text      (fn [s]
+                              (reset! full-name (string/trim s))
+                              (validate-name s))]
+    (fn []
+      [quo/overlay
+       {:type            :shell
+        :container-style (style/page-wrapper insets)}
+       [quo/page-nav
+        {:key        :header
+         :background :blur
+         :icon-name  :i/arrow-left
+         :on-press   #(rf/dispatch [:navigate-back])}]
+       [rn/keyboard-avoiding-view
+        {:key   :content
+         :style style/screen-container}
+        [rn/view {:style {:gap 22}}
+         [quo/text-combinations {:title (i18n/label :t/name)}]
+         [quo/input
+          {:theme           :dark
+           :blur?           true
+           :error?          (seq @error-msg)
```

Comment:
Another reason to be aware `seq` on strings is probably not what we want. Here `:error?` may contain a sequence of characters, which is more expensive for Reagent to compare. We actually want to pass a boolean, so better is `:error? (not (string/blank? @error-msg))` (see my comment below about benchmarks).